internal/chart/flags.go

```diff
@@ -1,6 +1,74 @@
 package chart
 
-func AppendContainersFlags(flags map[string]interface{}, publisherURL, traceCollectorURL, CPUUtilizationPercentage, requeueDuration, buildExecutorArgs, maxSimultaneousJobs, healthzLivenessTimeout, requestBodyLimitMb, timeoutSec string) map[string]interface{} {
+type FlagsBuilder interface {
+	Build() map[string]interface{}
+	WithControllerConfiguration(CPUUtilizationPercentage string, requeueDuration string, buildExecutorArgs string, maxSimultaneousJobs string, healthzLivenessTimeout string, requestBodyLimitMb string, timeoutSec string) *flagsBuilder
+	WithDefaultPresetFlags(defaultBuildJobPreset string, defaultRuntimePodPreset string) *flagsBuilder
+	WithOptionalDependencies(publisherURL string, traceCollectorURL string) *flagsBuilder
+	WithRegistryAddresses(registryAddress string, serverAddress string) *flagsBuilder
+	WithRegistryCredentials(username string, password string) *flagsBuilder
+	WithRegistryEnableInternal(enableInternal bool) *flagsBuilder
+	WithRegistryHttpSecret(httpSecret string) *flagsBuilder
+	WithNodePort(nodePort int64) *flagsBuilder
+}
+
+type flagsBuilder struct {
```

Comment:
Wouldn't be easier to create flags in this way:
```
flags := map[string]string
```
key is path, value is value?

Less code more readable code.

Reply:
The [Run](https://github.com/helm/helm/blob/v3.10.1/pkg/action/install.go#L183) method from the library we use expects map[string]interface{} type and I'm not sure how it would react when we pass string instead of (for example) int64 type.

Additionally, the main goal of this PR is to enable another PR where we decided to not add another `Append` method and improve a little main idea of building flags.

This PR mainly moves old code (from the `Append` functions) into new architecture. I did not refactor most of the code inside them so, for example, we still keep code like [this](https://github.com/kyma-project/serverless-manager/blob/1f77f189427463fab42ef29ae95665ea4111fe10/internal/chart/flags.go#L129-L145) there we construct preset configuration using type map[string]interface{}.

I think it would be a good idea to look deeper into these functions but not in this PR.

Reply:
I think that I didn't write it clearly, I would like builder to have logic like that:
```
fb.flags["aa.bb.cc"] = value
```
We agreed to implement it this way in another PR